### ISSUE TITLE
CoverflowAltTab Disabled by Cinnamon

### ISSUE THREAD

**Issue by hyph3n** [dmo60/CoverflowAltTab#61]
Today I noticed my system (Fedora 23/Cinnamon) wasn't able to alt-tab at all. When I pressed the key combination, it simply cleared my desktop of active windows instead of animating through the window selection process as usual. 

I tried to reload the CoverflowAltTab code (which was installed via spices--not directly from GitHub).

Following this, hitting alt-tab worked again, but it was just the default alt-tab behavior and not the Coverflow style. 

I went into Cinnamon's "Extensions" settings, and saw that Coverflow Alt-Tab was disabled with the message: "Problem loading - please check Looking Glass or your system's error log". (NOTE: This message is visible as a tool tip if you mouse over the red dot to the right of the extension name).

I checked Looking Glass, and I see the following messages (actual user name in paths below has been replaced with {user} for privacy):

```
error t=2016-05-25T19:44:48Z second argument to Lang.bind() must be a function, not undefined
trace t=2016-05-25T19:44:48Z 
<----------------
bind@resource:///org/gnome/gjs/modules/lang.js:81
KeybinderNewApi.prototype.disable@/home/{user}/.local/share/cinnamon/extensions/CoverflowAltTab@dmo60.de/keybinder.js:54
Manager.prototype.disable@/home/{user}/.local/share/cinnamon/extensions/CoverflowAltTab@dmo60.de/manager.js:46
disable@/home/{user}/.local/share/cinnamon/extensions/CoverflowAltTab@dmo60.de/extension.js:60
prepareExtensionUnload@/usr/share/cinnamon/js/ui/extensionSystem.js:32
unloadExtension@/usr/share/cinnamon/js/ui/extension.js:498
reloadExtension@/usr/share/cinnamon/js/ui/extension.js:530
Melange.prototype.ReloadExtension@/usr/share/cinnamon/js/ui/lookingGlass.js:661
_handleMethodCall@resource:///org/gnome/gjs/modules/overrides/Gio.js:261
_wrapJSObject/<@resource:///org/gnome/gjs/modules/overrides/Gio.js:331
---------------->
error t=2016-05-25T19:44:48Z [Extension "CoverflowAltTab@dmo60.de"]: Failed to evaluate 'disable' function on extension: CoverflowAltTab@dmo60.de
error t=2016-05-25T19:44:48Z [Extension "CoverflowAltTab@dmo60.de"]: Extension is not compatible with current Cinnamon version
error t=2016-05-25T19:44:48Z Could not load extension CoverflowAltTab@dmo60.de
```

I remember running a "dnf upgrade" either yesterday or the day before, so I suspect this is the reason for the "Extension is not compatible with current Cinnamon version" message. Let me know if you have any advice, or if there's anything you'd like me to test out. In the meantime, at least the default alt-tab behavior is working for me.

Thanks.

**Comment by p91paul** [comment]
Version on cinnamon spices is quite outdated unfortunately. Could you please test the github version? I should look into getting ownership of the extension on cinnamon spices as well.

**Comment by hyph3n** [comment]
Thanks for the reply, but I'm afraid the extension still doesn't seem to be working. Alt-tab clears my desktop of all active windows and slightly darkens my desktop background, but does not actually show me a list of windows. I performed the following steps:

1) Uninstalled cinnamon-spices version of the extension.
2) Cloned this repo.
3) Manually copied the folder "CoverflowAltTab@dmo60.de" to ~/.local/share/cinnamon/extensions
4) Clicked "Add to Cinnamon" in Settings->Extensions.
5) Alt+F2 -> r

After performing the above steps, here's what my desktop looks likewhile I'm alt-tabbing. As soon as I let go of the keys, my desktop returns to normal:

![alt-tab](https://cloud.githubusercontent.com/assets/5378617/15573001/212b6be4-2312-11e6-9a3b-e04321ac27d5.png)

As you can see, this is a dual desktop setup in case that's relevant. This time there were no errors in Looking Glass--I only see the following line related to CoverflowAltTab:

```
info t=2016-05-26T07:10:07Z Loaded extension CoverflowAltTab@dmo60.de in 6 ms
```
After navigating back to Settings->Extensions and removing CoverflowAltTab, I'm again able to alt-tab through my windows using the default Cinnamon behavior.

Thanks for any help you can offer!

**Comment by p91paul** [comment]
I should add this to the README, until a proper fix is issued. For whatever reason, on Cinnamon it only works if you enable the "Always show the animation on the primary desktop" extension option.
Please don't close this bug report (even if for you the workaround is ok), I will close if and when it is actually fixed.
Thanks for taking the time to report!

**Comment by hyph3n** [comment]
Yep, that workaround fixed things for me. Thanks!

**Comment by machitgarha** [comment]
#145

This issue should be closed.

**Comment by p91paul** [comment]
I merged #145 and decided to drop Cinnamon support.
This extension, or most of it at least, should be part of cinnamon anyway; I'm aware that this extension provided more options that the cinnamon builtin, but I'm not a cinnamon user and I'm unable to provide support for it due to limited time.
If anyone wants to contribute working cinnamon code, it will be welcome. Infrastructure to support multiple platforms should be ready in platform.js.
